flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/serde/ChangelogModeJsonDeserializer.java

```diff
@@ -49,8 +51,14 @@ public ChangelogMode deserialize(
         ChangelogMode.Builder builder = ChangelogMode.newBuilder();
         JsonNode rowKindsNode = jsonParser.readValueAsTree();
         for (JsonNode rowKindNode : rowKindsNode) {
-            RowKind rowKind = RowKind.valueOf(rowKindNode.asText().toUpperCase());
-            builder.addContainedKind(rowKind);
+            final String rowKindText = rowKindNode.asText().toUpperCase();
```

Comment:
```suggestion
            final String rowKindText = rowKindNode.asText().toUpperCase(Locale.ROOT);
```
what values are planned to be there, should we care. about something tricky e.g. turkish locales?

Reply:
we can also just remove this `toUpperCase` . IMHO `toUpperCase` / `toLowerCase` are just adding unnecessary runtime overhead and lead to weird bugs.